website/docs/getting-started/template-walkthrough.md

```diff
@@ -0,0 +1,142 @@
+
+Next, we initialize `app` which will be the root of our CDK Construct tree:
+
+```typescript
+const app = new App();
+```
+
+CDK models infrastructure as a tree of Constructs with a single instance of `App` at the root.
+To add Constructs to the tree, we pass the parent as the first argument when instantiating a child.
+
+This is exactly what we do next with `Stack`:
+
+```typescript
+const stack = new Stack(app, "MyStack");
+```
+
+`stack` will be the parent of the rest of the Constructs we create.
+In CDK, all Resources belong to a Stack - a Stack is a collection of Resources that are deployed and managed together.
+
+:::note
+
+An `App` may contain many stacks, but that is outside the scope of this document.
+You can read more about [`App`][cdk-app] and [`Stack`][cdk-stack] in the CDK documentation.
+
+:::
+
+Once we have an instance of `stack`, we're ready to attach our first `Function`:
+
+```typescript
+const sayFunction = new Function(
+  stack,
+  "SayFunction",
+  async (event: { message: string }) => {
+    // This code will run on AWS Lambda
+    console.log(event.message);
+    return;
+  }
+);
+```
+
+This might not be the most interesting code in the world, but it offers us our first glimpse at the magic of Functionless.
+
+If you've used CDK before, you're probably familiar with the seperation it imposes between infrastructure and application code.
+Functionless bridges this gap by letting you write application code inline with the infrastructure it sits atop.
+
+This opens up a world of new possibilities for how code and infrastructure can interact.
+We can see this in action with the `StepFunction` Construct definition:
+
+```typescript
+new StepFunction(stack, "Workflow", async (event: { name: string }) => {
+  // This will run in an AWS Step Function
+  await sayFunction({ message: `Hello ${event.name}` });
+});
+```
+
+There is a lot to unpack in these three lines.
+
+First off, we're able to call the `sayFunction` Construct we just defined as if it were a standard nodejs `function`.
+The types inferred from arrow function in the `Function` definition above are preserved, giving us type-checking when we call `sayFunction`.
+In Functionless, types are maintained across service boundaries, removing a whole class of user error.
+
+Secondly, Functionless introspects how your runtime code interacts with its infrastructure and uses this information to automatically configure least-privilege policies.
+
+:::note
+
+CDK happens to have equivalent functionality for inferring IAM permissions for Step Functions, but this behavior does not extend to other runtime usages such as Lambda.
+
+In Functionless, this ability extends to any runtime usage as we'll see later when we [modify the example](./change-the-code).
+
+:::
+
+Lastly, if you've ever created an AWS Step Function before, you might be wondering where all the Amazon States Language (ASL) went.
```

Comment:
Random deep dive of the day

https://www.grammarphobia.com/blog/2010/05/firstly-secondly-and-thirdly.html

> We think it’s OK to use either the long or short adverbs to make your points. But if you’re going to use the “ly” endings, we’d recommend “finally” instead of the awkward “lastly.”